lang/TUTORIAL_FRENCH.md

```diff
@@ -0,0 +1,676 @@
+
+Enfin, ne laissez pas les champs de logique métier affecter la forme générale de l'API.
+Les données du domaine métier restent le modèle de base. Si vous trouvez que la logique métier ne s'adapte pas vraiment, c'est le signe que votre modèle sous-jacent n'est peut-être pas le bon.
+
+### Cinquième étape : Mutations
+
+La dernière pièce manquante de notre conception du schéma GraphQL est la capacité à modifier réellement les valeurs : création, mise à jour et suppression des collections et des éléments connexes.
+Comme pour la partie lisible du schéma, nous devrions commencer par une vue de haut niveau : dans ce cas, des diverses mutations que nous voulons mettre en œuvre, sans nous soucier de leurs entrées ou sorties spécifiques. Naïvement, nous pourrions suivre le paradigme CRUD et n'avoir que des mutations `create`, `delete`, et `update`.
+Bien que ce soit un bon point de départ, c'est insuffisant pour une API GraphQL correcte.
+
+### Actions logiques séparées
+
+La première chose que nous pourrions remarquer si nous nous en tenions au CRUD est que notre mutation `update` devient rapidement massive, responsable non seulement de la mise à jour de simples valeurs scalaires comme le titre, mais aussi de l'exécution d'actions complexes comme la publication/dé-publication, l'ajout/suppression/réorganisation des produits dans la collection, la modification des règles pour les collections automatiques, etc. Cela rend l'implémentation difficile sur le serveur et le raisonnement difficile pour le client.
+Au lieu de cela, nous pouvons tirer parti de GraphQL pour le diviser en actions logiques plus granulaires. Dans un premier temps, nous pouvons séparer les opérations de publication et de dépublication, ce qui donne la liste de mutations suivante :
+- créer
+- supprimer
+- mettre à jour
+- publier
+- dépublier
+
+*Règle n°14 : écrire des mutations distinctes pour des actions logiques distinctes sur une ressource.*
+
+### Manipulation des relations
+
+La mutation `update` a encore beaucoup trop de responsabilités, il est donc logique de continuer à la diviser, mais nous allons traiter ces actions séparément car elles valent la peine d'être considérées sous une autre dimension : la manipulation des relations entre objets (par exemple, one-to-many, many-to-many). Nous avons déjà examiné l'utilisation des ID par rapport à l'intégration, et l'utilisation de la pagination par rapport aux tableaux dans l'API de lecture, et il y a des problèmes similaires à traiter lors de la modification de ces relations.
+
+En ce qui concerne la relation entre les produits et les collections, il existe plusieurs styles que nous pourrions envisager de manière générale :
+- Intégrer la relation entière (par exemple `produits : [EntréeProduit !]!`) dans la mutation de mise à jour est le style CRUD par défaut, mais bien sûr, cela devient rapidement inefficace lorsque la liste est grande.
```

Comment:
```suggestion
- Intégrer la relation entière (par exemple `products : [ProductEntry!]!`) dans la mutation de mise à jour est le style CRUD par défaut, mais bien sûr, cela devient rapidement inefficace lorsque la liste est grande.
```

Not sure what the proper types should be here but I don’t think we should translate API stuff to French